gmso/abc/abstract_site.py

```diff
@@ -24,6 +24,8 @@ def default_position():
 class Site(GMSOBase):
     __iterable_attributes__: ClassVar[set] = {
         "label",
+        "group",
+        "molecule",
```

Comment:
Molecule label will be a molecule name, and a molecule number in this case to differentiate exactly which molecule a site is grouped in.